packages/react-refresh/src/ReactFreshBabelPlugin.js

```diff
@@ -98,9 +98,20 @@ export default function (babel, opts = {}) {
             if (!foundInside) {
               return false;
             }
-            // const Foo = hoc1(hoc2(() => {}))
-            // export default memo(React.forwardRef(function() {}))
-            callback(inferredName, node, path);
+
+            // TODO: this is a hack, we should find a better way to detect this.
```

Comment:
I suck at babel, and this call is recursive. Consider this a proof of concept of a fix.